### PR TITLE
Port/refactor of f-m maintenance-mode tests from testfm

### PR DESCRIPTION
**Pull request by Gauravtalreja1**
Description:

1. Port/refactor f-m maintenance-mode subcommand tests from TestFM to robottelo
https://github.com/SatelliteQE/testfm/blob/master/tests/test_maintenance_mode.py

It includes changes from https://github.com/SatelliteQE/robottelo/pull/10227, and it will be rebased when either PR is merged

Signed-off-by: Gaurav Talreja <gauravtalreja1@gmail.com>

### CODE REVIEW

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/maintain/test_maintenance_mode.py

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/maintain/test_maintenance_mode.py

File: pytest_fixtures/component/maintain.py [hunk 1]
```diff
@@ -59,3 +58,35 @@ def module_synced_repos(sat_maintain):
...
+    sat_maintain.execute(f'cp {satellite_maintain_yml} foreman_maintain.yml')
+    sat_maintain.execute(f'sed -i "$ a :manage_crond: true" {satellite_maintain_yml}')
+
+    yield sat_maintain.api.SyncPlan(organization=org.label).search(query={'search': 'enabled=true'})
+
+    @request.addfinalizer
+    def _finalize():
+        assert sat_maintain.cli.MaintenanceMode.stop().status == 0
+        sat_maintain.execute(f'cp foreman_maintain.yml {satellite_maintain_yml}')
+        sat_maintain.execute(f'rm -rf {satellite_maintain_yml}')
```
Comment (vsedmik) [inline pytest_fixtures/component/maintain.py:87]: I think we want to remove the backup file here.
```suggestion
        sat_maintain.execute(f'rm -rf foreman_maintain.yml')
```

**Review by vsedmik — CHANGES_REQUESTED** [review]
Two small comments...

File: robottelo/constants/__init__.py [hunk 2]
```diff
@@ -1635,6 +1635,10 @@ class Colored(Box):
 HAMMER_CONFIG = "~/.hammer/cli.modules.d/foreman.yml"
 HAMMER_SESSIONS = "~/.hammer/sessions"
 
+satellite_answer_file = "/etc/foreman-installer/scenarios.d/satellite-answers.yaml"
+maintain_hammer_yml = "/etc/foreman-maintain/foreman-maintain-hammer.yml"
+satellite_maintain_yml = "/etc/foreman-maintain/foreman_maintain.yml"
```
[the remark this answers is not in the document]
Comment (Gauravtalreja1) [reply robottelo/constants/__init__.py]: @vsedmik Updated

File: pytest_fixtures/component/maintain.py (hunk 1 above)
Comment (Gauravtalreja1) [reply pytest_fixtures/component/maintain.py]: Silly miss, updated, @vsedmik request to re-review

File: robottelo/constants/__init__.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (vsedmik) [reply robottelo/constants/__init__.py]: @Gauravtalreja1 don't forget to update references in the other PRs.

**Review by vsedmik — APPROVED** [review]
ACK

File: robottelo/constants/__init__.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Gauravtalreja1) [reply robottelo/constants/__init__.py]: Sure sir

**Comment by jyejare** [comment]
@Gauravtalreja1 Please rebase the code and fix final comments/force_Resolve.

**Comment by jyejare** [comment]
Ping @Gauravtalreja1 !

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/maintain/test_maintenance_mode.py

**Review by jameerpathan111 — APPROVED** [review]

**Review by jyejare — APPROVED** [review]

**Pull request merged**